skyportal/handlers/api/source.py

```diff
@@ -249,6 +255,11 @@ def get(self, obj_id=None):
             )
             if s is None:
                 return self.error("Invalid source ID.")
+            if "ps1" not in [thumb.type for thumb in s.thumbnails]:
+                create_thumbnail_link = functools.partial(
+                    add_ps1_thumbnail_and_push_ws_msg, s, self,
+                )
+                IOLoop.current().add_callback(create_thumbnail_link)
```

Comment:
Maybe `functools.partial` is overkill here.  How about:

```
IOLoop.current().add_callback(
    lambda: add_ps1_thumbnail_and_push_ws_msg(s, self)
)
```

I was a bit concerned about using `self` (the handler) this way, because they handler is typically destroyed after each request.  But, I *think* you're okay since a reference to self, like you are installing here, might keep the handler alive.  I would test this explicitly by adding a `time.sleep(...)` to `create_thumbnail_link_and_push_ws_msg` (which, btw, I think is fine to rename to `create_thumbnail_link` if you use a lambda); if the sleep lasts a long time, it would give time for the handler to finish, and we would see if it still available when the WS message needs to be pushed.

Reply:
@stefanv confirmed, the handler stays alive even with a long sleep

Reply:
@stefanv regarding functools vs lambda, I'm not sure I see it as a simplification, but I've implemented your suggestion (now using a lambda function)